cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java

```diff
@@ -184,23 +183,24 @@ private void restoreActiveRuns() {
                   }
                   try {
                     LOG.info("Found active run: {}", runRecordDetail.getProgramRunId());
-                    if (runRecordDetail.getStatus() == ProgramRunStatus.PENDING) {
-                      runRecordMonitorService.addRequest(runRecordDetail.getProgramRunId());
-                    } else if (runRecordDetail.getStatus() == ProgramRunStatus.STARTING) {
-                      runRecordMonitorService.addRequest(runRecordDetail.getProgramRunId());
+                    ProgramOptions programOptions =
+                        new SimpleProgramOptions(
+                            runRecordDetail.getProgramRunId().getParent(),
+                            new BasicArguments(runRecordDetail.getSystemArgs()),
+                            new BasicArguments(runRecordDetail.getUserArgs()));
+                    ProgramDescriptor programDescriptor = this.store.loadProgram(
+                        runRecordDetail.getProgramRunId().getParent());
+                    if (runRecordDetail.getStatus() == ProgramRunStatus.STARTING) {
+                      flowControlService.addRequest(runRecordDetail.getProgramRunId(),
```

Comment:
I don't think you need this. Run is already in the database.

Reply:
Removed.